### PR TITLE
[fastlane_core] Fix project_paths() in project.rb to respect `container:` references

### PR DESCRIPTION
**Pull request by svendr**
### Checklist
- [x] I've run `bundle exec rspec` from the root directory to see all new and existing tests pass
- [x] I've followed the _fastlane_ code style and run `bundle exec rubocop -a` to ensure the code style is valid
- [x] I've read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I've updated the documentation if necessary.

### Motivation and Context
As stated in #13660 and #11034, paths to xcodeprojects can't be resolved, if the projects are referenced `Relative to Workspace`(= `container:`) instead of `Relative to Group` (= `group:`), which results in Gym's `CodeSigningMapper` not being able to detect the correct provisioning profile mapping.

There are other edge cases of project-references to think of:
* types `absolute:` (= `Absolute Path`) and `developer:` (= `Relative to Developer Dir`): both valid, but should be considered bad practice and therefore not really relevant
* nested, group:-referenced .xccodeproj files: uncommon but possible (a .xcodeproj file inside a folder referenced by group in a workspace)

### Description
This PR fixes support for top-level `container:`- and `absolute:`-references by switching from Regex-based parsing of the `contents.xcworkspacedata` file to Xcodeproj-Gem-based parsing (which traverses the XML properly).

Xcodproj provides a convenient way to parse all top-level [file-references](https://www.rubydoc.info/github/CocoaPods/Xcodeproj/Xcodeproj/Workspace#file_references-instance_method) of a workspace. And the workspace-property of `FastlaneCore::Project` is already a `Xcodeproj::Workspace` :)

So I've updated the `project_paths` method in `project.rb` accordingly.

**Tests:** I've updated `project_spec.rb` to validate the handling of `group:` and `container:` references properly. Added some Example-Workspaces for this too. `absolute:` references aren't considered here as CI would fail for obvious reasons.

### CODE REVIEW

**Comment by svendr** [comment]
Just signed the CLA

**Review by getaaron — APPROVED** [review]
Looks like a good change, thanks for the PR! Would like to get a second approval (maybe from @joshdholtz or @ohayon?) before merging

File: fastlane_core/lib/fastlane_core/project.rb
Comment (svendr) [reply fastlane_core/lib/fastlane_core/project.rb]: @getaaron Thanks for your feedback and great idea, just committed that!

**Review by joshdholtz — APPROVED** [review]
Thanks for making these changes! Moving things over to `xcodeproj` is 💯

**Pull request merged**
